openshift/backplane-cli
# OSD-19438 Add a troubleshoot subcommand to print debug info

### What type of PR is this?

feature

### What this PR does / Why we need it?
This is a draft to show the idea, refinement is needed. 

This PR adds a `troubleshoot` sub-command to print debug info of the current environment.

We usually receive asks from users that "something is not working" and turns out the environment is mis-configured.

There's a gap between "how the user think the environment is" and "how backplane-cli think the environment is". This PR aims to show "how backplane-cli think the environment is" to assist troubleshooting.

Sample output:
```
$ ocm backplane troubleshoot
[V] backplane-cli configuration path: /Users/feichashao/.config/backplane/config.stg.json
[V] proxy in backplane-cli config: http://example.com:8888
[V] backplane url from OCM: https://backplane-api.example.com
[V] oc server url: https://backplane-api.example.com/backplane/cluster/2876r9d2b449edkaq7g95sijoc1k135e/
[V] proxy in OC configuration: http://example.com:8888
To verify proxy connectivity, run:
 HTTPS_PROXY=http://example.com:8888 curl -Iv https://www.redhat.com 
To verify backplane-api connectivity, run:
 HTTPS_PROXY=http://example.com:8888 curl -Iv https://backplane-api.example.com
```

### Which Jira/Github issue(s) does this PR fix?
https://issues.redhat.com/browse/OSD-19438

### Special notes for your reviewer

### Pre-checks (if applicable)
